WooCommerce/Classes/ViewRelated/Products/ProductsViewController.swift

```diff
@@ -969,15 +975,41 @@ private extension ProductsViewController {
                     if let currentSelectedIndexPath {
                         tableView.deselectRow(at: currentSelectedIndexPath, animated: false)
                     }
-                    let scrollPosition: UITableView.ScrollPosition = tableView.indexPathsForVisibleRows?.contains(selectedIndexPath) == true ?
-                        .none: .middle
+
+                    let scrollPosition: UITableView.ScrollPosition = {
+                        let hasSelectedProductChanged = (selectedProduct != previous?.0)
+                        let isSelectedIndexPathVisible = self.tableView.indexPathsForVisibleRows?.contains(selectedIndexPath) == true
```

Comment:
Since we use this further down in `ProductsViewController.listenToSelectedProductToAutoScrollWhenProductChanges(product:)`, perhaps it should be a function?

```
 func isIndexPathVisible(_ indexPath: IndexPath) -> Bool {
        guard let indexPathsForVisibleRows = tableView.indexPathsForVisibleRows else {
            return false
        }
        return indexPathsForVisibleRows.contains(indexPath)
    }
```

Reply:
Thanks for the suggestion! Updated in https://github.com/woocommerce/woocommerce-ios/pull/12060/commits/5adf6fb1319bf2277de8423cbe0c4d056ee261e6 for this and the following 2 comments